cpp/src/parquet/page_index.cc

```diff
@@ -88,9 +88,8 @@ class TypedColumnIndexImpl : public TypedColumnIndex<DType> {
  public:
   using T = typename DType::c_type;
 
-  TypedColumnIndexImpl(const ColumnDescriptor& descr,
-                       const format::ColumnIndex& column_index)
-      : column_index_(column_index) {
+  TypedColumnIndexImpl(const ColumnDescriptor& descr, format::ColumnIndex column_index)
+      : column_index_(std::move(column_index)) {
```

Comment:
Are there other places in Parquet C++ where a similar improvement can be done?

Reply:
Actually I just find some remarkable place have similar improvement.

It seems that Parquet C++ metadata system uses some hacking, the thrift is deserialized once, and use pointer like `void*` to pass it to other component, e.g.:

```
FileMetaData <-- will deserialize thrift, and store as member `metadata_`
RowGroupMetadata: uses reference from `FileMetadata.metadata_`
ColumnChunkMetadata: uses reference from parent...
```

This style make sure the performance ( though tricky ) without copying `ContainerT<format::Type>`

Reply:
But I think if we have many row-groups and column chunks, during building, it can using `move` instead of `copy` to avoid some memcpy